pkg/client/restclient/request.go

```diff
@@ -910,12 +910,26 @@ func (r *Request) transformResponse(resp *http.Response, req *http.Request) Resu
 		return Result{err: errors.FromObject(status)}
 	}
 
-	// TODO: Check ContentType.
+	contentType := resp.Header.Get("Content-Type")
+	var decoder runtime.Decoder
+	if contentType == r.content.ContentType {
```

Comment:
You mean that I should have:

else {
  mediaType, params, err := mime.Parse(contentType)
  if err != nil {
    return Result{err: "error parsing content type"}
  }
  decoder, err := r.serializers.RenegotiatedDecoder(mediaType, params)
  if err != nil {
    // return what I have
  }
}

or sth else?